dev-tools/dep-info.js

```diff
@@ -0,0 +1,123 @@
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+const { join } = require('path');
+const { readFileSync, writeFileSync, readdirSync } = require('fs');
+const { request } = require('undici');
+
+const ROOT_DIR = join(__dirname, '../');
+const ROOT_NODE_MODULES = join(ROOT_DIR, 'node_modules');
+const CHROMIUM_LICENSE_URL =
+  'https://chromium.googlesource.com/chromium/src/+/refs/heads/main/LICENSE?format=TEXT';
+
+function safeReadDirSync(directory) {
+  let files = [];
+  try {
+    files = readdirSync(directory);
+  } catch (_) {
+    // do nothing
+  }
+  return files;
+}
+
+function getFileContent(directory, filename) {
+  const files = safeReadDirSync(directory);
+
+  const file = files.find(f => f.toLowerCase().startsWith(filename.toLowerCase()));
+  let content;
+  if (file) {
+    content = readFileSync(join(directory, file), 'utf8');
+  }
+  return content;
+}
+
+function generateDependencyInfo(deps) {
+  var allLicenses = [];
+  deps.forEach(d => {
+    const modulesPath = join(ROOT_NODE_MODULES, d);
+
+    const dep = { name: d };
+
+    const license = getFileContent(modulesPath, 'LICENSE');
+    if (license) {
+      dep.license = license;
+    }
+
+    const notice = getFileContent(modulesPath, 'NOTICE');
+    if (notice) {
+      dep.notice = notice;
+    }
+
+    allLicenses.push(dep);
+  });
+  return allLicenses;
+}
+
+function getMeta() {
+  const { name, dependencies = {} } = JSON.parse(
+    readFileSync(join(ROOT_DIR, 'package.json'), 'utf8')
+  );
+  return { name, dependencies };
+}
+
+async function getChromiumInfo() {
+  const { statusCode, body } = await request(CHROMIUM_LICENSE_URL);
+  const bodyStr = await body.text();
+  if (statusCode !== 200) {
+    throw new Error(
+      `Failed to fetch chromium license info. status: ${statusCode}, reason: ${bodyStr}`
+    );
+  }
+  const license = Buffer.from(bodyStr, 'base64').toString('utf-8');
+  return { name: 'chromium', license };
+}
+
+async function generateNotice() {
+  const { name: pkgName, dependencies } = getMeta();
+  const chromiumInfo = await getChromiumInfo();
+  const depInfo = [chromiumInfo, ...generateDependencyInfo(Object.keys(dependencies))];
```

Comment:
At first, I wanted to generate the NOTICE.txt file that includes only npm production dependencies, then generate the extended NOTICE.txt including Chromium license when building the app. But I figured we might simplify it by keeping only one NOTICE.txt in the repo and copy it over when we release the app.

WDYT @justinkambic @vigneshshanmugam ?

Reply:
I'll defer to @vigneshshanmugam as to whether we should have the chromium license included, but I am fine with bundling that all together when we release the app. With the size of the binary we're going to ship, including a full browser, some extra lines of license copy are not going to move the needle very much on dist size. That's my impression, at least.

Reply:
Oh let me clarify my intention. My initial approach was:
1. We keep `NOTICE.txt` listing npm production deps in the repo
2. When we release the app, add the chromium license to the existing `NOTICE.txt`. Now we have created (temp)NOTICE.txt
3. Copy over the (temp)NOTICE.txt as a build resource

and my question was that instead of taking step 2, we add the chromium license info in step 1 and remove step 2 completely(That's what I did in this PR too).

But now that I think about it, my initial approach doesn't really make sense as we don't need to make a distinction between node modules and chromium.

Reply:
++ Just keeping a single Notice.txt would be better IMO.